test/e2e/framework.go

```diff
@@ -419,3 +420,136 @@ func kubectlExec(namespace string, podName, containerName string, args ...string
 func KubeDescribe(text string, body func()) bool {
 	return Describe("[k8s.io] "+text, body)
 }
+
+// PodStateVerification represents a verification of pod state.
+// Any time you have a set of pods that you want to operate against or query,
+// this struct can be used to declaratively identify those pods.
+type PodStateVerification struct {
+	// Optional: only pods that have k=v labels will pass this filter.
+	selectors map[string]string
```

Comment:
per tims update above.  

- i think there is value in this, 
- but after hacking around with it for a little while, though, i think it should be done in a later iteration.
- why? as-is, the separation between the zeroOrderPredicates and verify felt a little awkward after i implemented it into the API.  It added a little indirection.  

i think we should have a full blown generic state definition for kube, and this will be the seed for it , but i don't think this PR is the place to design the final product....  thoughts @timothysc ?